### PR TITLE
Prevent software from spawning while disassembling usb drive with it

### LINKED ISSUE #14750: [Bug(?)] Misc. Software from disassembling USB drive
So I disassembled a USB drive with some misc software on it.. got the software as a physical object. Proceeded to smack a table to pieces with it, however it damaged the software. I'd like to find the creator of that software and shake his hand.

No but seriously, just disassemble a USB drive with misc software to replicate the problem. :)

### LINKED ISSUE #15384: Software once again can drop as an item
Not sure when did the regression happen, but dismantling USBs with software once again allows having software in inventory, on the map etc.

### PR DESCRIPTION
**Pull request by Night-Pryanik**
Closes #14750.
Closes #15384.

### CODE REVIEW

**Comment by BevapDin** [comment]
This should probably be in `player::complete_disassemble`. The activity can be canceled, which should keep the software.

**Comment by Coolthulhu** [comment]
Segfault. Most likely due to fact that `contents[0]` doesn't need to exist.

Also, this doesn't fully solve the issue of software spawns. You can still get itemized software by destroying an USB with fire or by bashing a wall with it until it breaks.

File: src/crafting.cpp [hunk 1]
```diff
@@ -1541,6 +1541,12 @@ void player::complete_disassemble( int item_pos, const tripoint &loc,
     if( veh != nullptr ) {
         veh_part = veh->part_with_feature(veh_part, "CARGO");
     }
+    
+    // If we're trying to disassemble usb drive with some software in it
+    // Erase contents of the usb drive to prevent spawning of software as item in inventory
+    if ( dis_item.contents[0].is_software() ) {
```
Comment (codemime) [inline src/crafting.cpp]: This one will probably cause crash if `contents` is empty.

**Comment by Night-Pryanik** [comment]
@Coolthulhu when did you get segfault? I got it compiled ok, ran the game ok, disassembled usb drive with software ok.
This PR is only about usb drive disassembling issues, not all software-itemization-related issues.

**Comment by Coolthulhu** [comment]
When I disassembled an item with no contents, in this case an USB drive.

**Pull request merged**